### PR TITLE
Move zstdgrep and zstdless tests to cli-tests

### PR DESCRIPTION
**Pull request by binhdvo**

### CODE REVIEW

File: tests/cli-tests/cltools/zstdgrep.sh.exit [hunk 1]
```diff
@@ -0,0 +1 @@
+1
```
Comment (Cyan4973) [inline tests/cli-tests/cltools/zstdgrep.sh.exit:1]: Is this file intentional ?

Comment (terrelln) [reply tests/cli-tests/cltools/zstdgrep.sh.exit:1]: Yup, it is saying that the exit code should be 1.

File: tests/cli-tests/cltools/zstdgrep.sh.stderr.glob [hunk 2]
```diff
@@ -0,0 +1 @@
+zstd: can't stat bad.zst : No such file or directory -- ignored
```
Comment (terrelln) [inline tests/cli-tests/cltools/zstdgrep.sh.stderr.glob]: nit: This could be `zstdgrep.sh.stderr.exact`.

If you run with `./run.py --preserve`, you can `cp scratch/cltools/zstdgrep.sh/stderr cltools/zstdgrep.sh.stderr.exact`

File: tests/cli-tests/cltools/zstdless.sh.stderr.glob [hunk 3]
```diff
@@ -0,0 +1,2 @@
+zstd: can't stat bad.zst : No such file or directory -- ignored
+bad.zst: No such file or directory
```
Comment (terrelln) [inline tests/cli-tests/cltools/zstdless.sh.stderr.glob]: nit: this could also be a `.exact` file.

**Review by terrelln — APPROVED** [review]
